Remove ingest processor supports excluding fields (#6244)
opensearch-project/documentation-website: _ingest-pipelines/processors/remove.md

| Parameter  | Required/Optional  | Description  |
|---|---|---|

> `field`  | Optional  | The field name containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields cannot be removed: `_index`, `_version`, `_version_type`, and `_id`. If a version is specified for the ingesting document, the `_id` cannot be removed. |

Reviewer: As written, "ingesting document" implies that the document is ingesting something. Is this accurate, or should we reword?

Reviewer 2: Revised: `The metadata fields `_index`, `_version`, `_version_type`, and `_id` cannot be removed. If `version` is specified, `_id` cannot be removed from the ingested document.`

> `field`  | Optional  | The name of the field containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields are not allowed to be removed: `_index`, `_version`, `_version_type`, and `_id`. Note that `_id` is not allowed to be removed when there's a specified external version for the ingesting document. |

Reviewer 2: ```suggestion
`field`  | Optional  | The field name containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields cannot be removed: `_index`, `_version`, `_version_type`, and `_id`. If a version is specified for the ingesting document, the `_id` cannot be removed. |
```

> `exclude_field`  | Optional  | The name of the fields need to be kept while other fields except the metadata fields will be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). `exclude_field` and `field` are mutually exclusive, either of them can be set. |

Reviewer 2: ```suggestion
`exclude_field`  | Optional  | The field name to retained. All other fields, except metadata fields, will be removed. The `exclude_field` and `field` options are mutually exclusive. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).  |
```
